message/message-util/util/src/java/org/sakaiproject/message/util/BaseMessage.java

```diff
@@ -1164,38 +1143,32 @@ && unlockCheck(function, reference)))
 
 			// get a list of the group refs, which are authzGroup ids
 			Collection<String> groupRefs = new Vector<String>();
-			for (Iterator<Group> i = groups.iterator(); i.hasNext();)
-			{
-				Group group = (Group) i.next();
-				groupRefs.add(group.getReference());
-			}
+            for (Group group : groups) {
+                groupRefs.add(group.getReference());
+            }
```

Comment:
```suggestion
            Collection<String> groupRefs = groups.stream().map(Group::getReference).collect(Collectors.toSet());
```

Reply:
Maybe that won't work. I'm guessing you tried that already.

Reply:
Do we need to use Vector here? Didn't we use that in the past for its concurrent access safety?

Reply:
my take was Vector use was just old-style sakai. as we're just fetching content, use of ArrayList seems preferred to Vector